server/controllers/stock/core.js

```diff
@@ -269,3 +281,231 @@ function stockManagementProcess(inventories) {
+ */
+function getStockConsumptionAverage(periodId, periodDate, numberOfMonths) {
+  const baseAvgNumberOfMonths = numberOfMonths || BASE_NUMBER_OF_MONTHS;
+
+  const baseDate = periodDate ? moment(periodDate).format(DATE_FORMAT) : moment().format(DATE_FORMAT);
+
+  const queryPeriodRange = `
+    SELECT id FROM period WHERE id BETWEEN ? AND ?;
+  `;
+
+  const queryPeriodId = periodId ?
+    'SELECT id FROM period WHERE id = ? LIMIT 1;' :
+    'SELECT id FROM period WHERE DATE(?) BETWEEN DATE(start_date) AND DATE(end_date) LIMIT 1;';
+
+  const queryStockConsumption = `
+    SELECT ROUND(AVG(s.quantity)) AS quantity, BUID(i.uuid) AS uuid, i.text, i.code, BUID(d.uuid) AS depot_uuid, d.text AS depot_text
+    FROM stock_consumption s
+    JOIN inventory i ON i.uuid = s.inventory_uuid 
+    JOIN depot d ON d.uuid = s.depot_uuid
+    JOIN period p ON p.id = s.period_id
+    WHERE p.id IN (?) 
+    GROUP BY i.uuid, d.uuid
+  `;
+
+  return db.one(queryPeriodId, [periodId || baseDate])
+    .then((period) => {
+      const beginingPeriod = period.id - baseAvgNumberOfMonths;
+      const paramPeriodRange = beginingPeriod > 0 ? [beginingPeriod + 1, period.id] : [1, period.id];
+      return db.exec(queryPeriodRange, paramPeriodRange);
+    })
+    .then((rows) => {
+      const ids = rows.map(row => row.id);
+
+      return db.exec(queryStockConsumption, [ids]);
+    });
+}
+
+/**
+ * Inventory Quantity and Consumptions
+ */
+function getInventoryQuantityAndConsumption(params) {
+  const bundle = {};
+  let status;
+  let delay;
+  let purchaseInterval;
+
+  if (params.status) {
+    status = params.status;
+    delete params.status;
+  }
+
+  if (params.inventory_delay) {
+    delay = params.inventory_delay;
+    delete params.inventory_delay;
+  }
+
+  if (params.purchase_interval) {
+    purchaseInterval = params.purchase_interval;
+    delete params.purchase_interval;
+  }
+
+  const sql = `
+    SELECT BUID(l.uuid) AS uuid, l.label, l.initial_quantity, 
+        SUM(m.quantity * IF(m.is_exit = 1, -1, 1)) AS quantity, 
+        d.text AS depot_text, l.unit_cost, l.expiration_date, 
+        BUID(l.inventory_uuid) AS inventory_uuid, BUID(l.origin_uuid) AS origin_uuid, 
+        l.entry_date, i.code, i.text, BUID(m.depot_uuid) AS depot_uuid,
+        i.avg_consumption, i.purchase_interval, i.delay,
+        iu.text AS unit_type
+    FROM stock_movement m 
+    JOIN lot l ON l.uuid = m.lot_uuid
+    JOIN inventory i ON i.uuid = l.inventory_uuid
+    JOIN inventory_unit iu ON iu.id = i.unit_id 
+    JOIN depot d ON d.uuid = m.depot_uuid 
+  `;
+
+  const clause = ' GROUP BY l.inventory_uuid, m.depot_uuid ';
+
+  return getLots(sql, params, clause)
+    .then((rows) => {
+      bundle.inventories = rows;
+      return getStockConsumptionAverage(null, params.dateTo);
+    })
+    .then((rows) => {
+      var sameInventory;
+      var sameDepot;
+
+      bundle.consumptions = rows;
+
+      for (let i = 0; i < bundle.consumptions.length; i++) {
```

Comment:
`bundle.consumptions` contains consumptions data which come from a function used by  others functions, the goal here is to get consumption information and current quantities for inventories; both come from two complex functions : `getLots()` and `getStockConsumptionAverage()`